pkg/http/request_log.go

```diff
@@ -156,7 +157,15 @@ func (h *RequestLogHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 
 func (h *RequestLogHandler) write(t *template.Template, in *RequestLogTemplateInput) {
-	if err := t.Execute(h.writer, in); err != nil {
+	// Use a local buffer to store the whole template expansion first. If h.writer
+	// is used directly, parallel template executions may result in interleaved
+	// output.
+	w := &bytes.Buffer{}
+	err := t.Execute(w, in)
+	if err == nil {
+		_, err = h.writer.Write(w.Bytes())
+	}
+	if err != nil {
 		// Template execution failed. Write an error message with some basic information about the request.
```

Comment:
Eh, I don't like the flow and if `h.writer.Write` fails, then `fmt.Fprintf` into the same writer is bound to fail again (most likely disk is dead)
So, I'd suggest the following flow:

```
if err := t.Execute(w, in); err != nil {
  fmt.Fprintf(...)
  return
}
h.writer.Write(...)
```

Alternative being, threading in logger and processing the writer error.
Note, that we were ignoring the error already from the Fprintf().


  
}

Reply:
Done